erdera/mapping_cnag_to_rd3.py

```diff
@@ -352,24 +445,77 @@ def build_import_phenotype_observations(client, data:pd.DataFrame):
                         new_entry['excluded'] = not observation.get('observed')
                         pheno_observations2.append(new_entry)
                         # save the phenotype names with code as prevalent in GPAP
-                        obs_dict[observation.get('name')] = observation.get('id').split(':')[1] # only get code (without HP:)
+                        obs_dict[observation.get('name')] = observation.get(
+                            'id')  # only get code (without HP:)
 
-    # convert phenotypic observations list to df 
+    # convert phenotypic observations list to df
     phen_observations = pd.DataFrame(pheno_observations2)
-    
+
     # map the phenotypic features from GPAP format to RD3
-    phen_dict = match_phenotypes(obs_dict)[0]
+    phen_dict, unmatched = asyncio.run(match_ontologies('Phenotypes', obs_dict))
     phen_observations['type'] = phen_observations['type'].replace(phen_dict)
 
     # get the unmatched phenotypes (the phenotypes from GPAP not present in RD3)
-    unmatched = match_phenotypes(obs_dict)[1]
-    unmatched_names = [entry.split(" (")[0] for entry in unmatched] # only get the name without the HP code
-    tmp = phen_observations.loc[phen_observations['type'].isin(unmatched_names)].index # get the indices of the rows to remove (no match)
-    phen_observations = phen_observations.drop(tmp, axis=0) # remove rows without a RD3 phenotype equivalent
+    # only get the name without the HP code
+    unmatched_names = [entry['invalid name'] for entry in unmatched]
+    tmp = phen_observations.loc[phen_observations['type'].isin(
+        unmatched_names)].index  # get the indices of the rows to remove (no match)
+    # remove rows without a RD3 phenotype equivalent
+    phen_observations = phen_observations.drop(tmp, axis=0)
 
     # upload
-    client.save_schema(table = 'Phenotype observations', data=phen_observations.drop_duplicates())
+    client.save_schema(table='Phenotype observations',
+                       data=phen_observations.drop_duplicates())
+
+    
+def build_import_variants(client: Client, data: pd.DataFrame):
+    """Builds and import variant information from GPAP to RD3
+    # disabled for now -- unfinished"""
+
+    variant_interpretations = data[[
+        'inheritance'
+    ]]
+
+    inheritance_dict = {
+        'HP:0001450': 'Y-linked inheritance',
+        'HP:0003745': '', # is sporadic inheritance, not included in our ontology
+        'HP:0000006': 'autosomal dominant inheritance',
+        'HP:0001419': 'X-linked recessive inheritance',
+        'HP:0001427': 'mitochondrial inheritance',
+        'HP:0001444': '', # 'typified by somatic mosaicism' not in included in our ontology
+        'HP:0000007': 'autosomal recessive inheritance',
+        'HP:0001470': 'sex-limited autosomal recessive inheritance, sex-limited autosomal dominant inheritance', #? the HP code is 'Sex-limited expression'
+        'HP:0001417': 'X-linked inheritance'
+    }

```

Comment:
Note for future: maybe we could move this to the ontology mappings schema when available